iceoryx2-ffi/ffi/src/api/service_builder_pub_sub.rs

```diff
@@ -152,10 +155,114 @@ impl IntoCInt for PublishSubscribeOpenOrCreateError {
     }
 }
 
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub enum iox2_type_variant_e {
+    FIXED_SIZE,
+    DYNAMIC,
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub enum iox2_type_detail_error_e {
+    INVALID_TYPE_NAME = IOX2_OK as isize + 1,
+    INVALID_TYPE_VARIANT_VALUE,
+    INVALID_SIZE_OR_ALIGNMENT_VALUE,
+}
+
 // END type definition
 
 // BEGIN C API
 
+/// Sets the max publishers for the builder
+///
+/// # Arguments
+///
+/// * `service_builder_handle` - Must be a valid [`iox2_service_builder_pub_sub_ref_h`]
+///   obtained by [`iox2_service_builder_pub_sub`](crate::iox2_service_builder_pub_sub) and
+///   casted by [`iox2_cast_service_builder_pub_sub_ref_h`](crate::iox2_cast_service_builder_pub_sub_ref_h).
+/// * `type_variant` - The [`iox2_type_variant_e`] for the payload
+/// * `type_name_str` - Must string for the type name.
+/// * `type_name_len` - The length of the type name string, not including a null
+/// * `size` - The size of the payload
+/// * `alignment` - The alignment of the payload
+///
+/// Returns IOX2_OK on success, an [`iox2_type_detail_error_e`] otherwise.
+///
+/// # Safety
+///
+/// * `service_builder_handle` must be valid handles
+/// * `type_name_str` must be a valid pointer to an utf8 string
+/// * `size` and `alignment` must satisfy the Rust `Layout` type requirements
+#[no_mangle]
+pub unsafe extern "C" fn iox2_service_builder_pub_sub_set_payload_type_details(
+    service_builder_handle: iox2_service_builder_pub_sub_ref_h,
+    type_variant: iox2_type_variant_e,
+    type_name_str: *const c_char,
+    type_name_len: c_size_t,
+    size: c_size_t,
+    alignment: c_size_t,
+) -> c_int {
+    debug_assert!(!service_builder_handle.is_null());
+    debug_assert!(!type_name_str.is_null());
+
+    let type_name = slice::from_raw_parts(type_name_str as _, type_name_len as _);
+
+    let type_name = if let Ok(type_name) = str::from_utf8(type_name) {
+        type_name.to_string()
+    } else {
+        return iox2_type_detail_error_e::INVALID_TYPE_NAME as c_int;
+    };
+
+    match type_variant as usize {
+        0 => (),
+        1 => (),
+        _ => return iox2_type_detail_error_e::INVALID_TYPE_VARIANT_VALUE as c_int,
```

Comment:
Should the handing here be the same as with the service type? It's also a situation where an int would have been used instead of the enum